app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt

```diff
@@ -38,19 +40,14 @@ class CountryDetailsFragment : Fragment(), CountryDetailsAdapter.ViewHolder.OnIt
         recyclerView = rootView.detailsRecyclerView
         adapter = CountryDetailsAdapter(emptyList(), this)
         recyclerView.adapter = adapter
+        (activity?.application as CountriesApplication).countriesComponent.inject(this)
         return rootView
     }
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         val id = arguments?.get(COUNTRY_ID) as? String
-        val viewModel: CountryDetailsViewModel by viewModels {
-            CountryDetailsViewModelFactory(
-                CountriesApi(
-                    apolloClient(requireContext())
-                )
-            )
-        }
+        val viewModel: CountryDetailsViewModel by viewModels { viewModelFactory }
```

Comment:
Let's move this to class level property.

Reply:
Done.